app/parse.py

```diff
@@ -15,8 +83,108 @@ class Product:
     num_of_reviews: int
 
 
+def fetch_page(url: str, driver_manager: WebDriverManager) -> BeautifulSoup:
+    driver_manager.navigate_to(url)
+    cookie_accept(url, driver_manager)
+    while True:
+
+        more_button = driver_manager.driver.find_elements(By.CLASS_NAME, "ecomerce-items-scroll-more")
+
+        if not more_button or \
+                more_button[0].value_of_css_property("display") == "none":
+
+            print("No more 'More' button (display: none).")
+            break
+
+        try:
+
+            driver_manager.click_element(
+                By.CLASS_NAME,
+                "ecomerce-items-scroll-more"
+            )
+            print("Clicked 'More' button.")
+            time.sleep(1)
+
+        except NoSuchElementException:
+            print("No more 'More' button.")
+            break
+        except Exception as e:
+            print(f"An error occurred: {e}")
+            break
+    page_source = driver_manager.get_page_source()
+
+    return BeautifulSoup(page_source, "html.parser")
+
+
+def cookie_accept(url: str, driver_manager: WebDriverManager) -> None:
+    driver_manager.navigate_to(url)
+    driver_manager.click_element(
+        By.XPATH,
+        '//*[@id="cookieBanner"]/div[2]/button'
+    )
+    print("Cookie accepted.")
+    
+
+def parse_single_product(soup: Tag) -> Product:
+    title = soup.find(class_="title").text.strip()
+    description = soup.find(class_="description").text.strip()
+    price = float(soup.find(class_="price").text.strip().replace("$", ""))
+    rating = len(soup.find_all("span", class_="ws-icon ws-icon-star"))
+    num_of_reviews = int(soup.find(class_="review-count").text.split()[0])
+    print(f"title: {title}")
+    return Product(
+        title=title,
+        description=description,
+        price=price,
+        rating=rating,
+        num_of_reviews=num_of_reviews
+    )
+
+
+def extract_all_products(soup: Tag) -> Product:
+    all_products = soup.find_all('div', class_='col-md-4 col-xl-4 col-lg-4')
+    print(f"Found {len(all_products)} products")
+    products_list = []
+    for product in all_products:
+        products_list.append(parse_single_product(product))
+    return products_list
+
+
+def write_to_csv(file_name: str, products: list[Product]) -> None:
+    with open(file_name, mode="w", newline="", encoding="utf-8") as csv_file:
+        writer = csv.writer(csv_file)
+        writer.writerow([
+            "Title",
+            "Dscription",
```

Comment:
There is a typo in the CSV header: 'Dscription' should be 'Description'. This will ensure that the CSV file has the correct headers.